scripts/logbook.py

```diff
@@ -240,44 +746,18 @@ def main(lims, pid, epp_logger):
             else:
                 instrument_name = pro.instrument.name
                 log.append(instrument_name)
-                if record["details"][instrument_number] is not '':
-                    for item in record["details"][instrument_number]:
-                        try:
-                            udf_content = pro.udf[item[4:]]
-                            udf_detail.append(item[4:]+":"+pro.udf[item[4:]])
-                        except KeyError, e:
-                            continue
-                    if udf_detail == []:
-                        log.append("-")
-                    else:
-                        log.append(','.join(udf_detail))
-                    write_record(log,record["dest_file"][instrument_number])
-                else:
-                    log.append("-")
-                    write_record(log,record["dest_file"][instrument_number])
+                details = get_details(record[instrument],pro)
+                log.append(details)
+                write_record(log,record[instrument]["dest_file"])
 
         elif instrument[0:3] == "udf":
             try:
                 instrument_name = pro.udf[instrument[4:]]
-                if pro.udf[instrument[4:]] is not '':
+                if pro.udf[instrument[4:]] != '':
```

Comment:
Since you have already assigned `pro.udf[instrument[4:]]` to `instrument_name`, maybe can use that here

```
if instrument_name:
    log.append(instrument_name)
```